horovod/tensorflow/__init__.py

```diff
@@ -1048,7 +1047,9 @@ def PartialDistributedGradientTape(gradtape, device_dense='', device_sparse='',
 
         The rest of the arguments are similar to those of DistributedGradientTape.
         """
-        if local_layers:
+        if local_layers is None:
+            local_layers = []
+        else:
```

Comment:
One final suggestion to simplify the code a bit: We don't really need multiple hierarchies of `if` branches here.
```
if local_layers is None:
  ...
elif isinstance(...
  ...
elif not all(isinstance(...
 ...
```
should work